src/index.js

```diff
@@ -359,11 +359,15 @@ async function main(req, context) {
     ));
   }
   const results = flatten(responses);
-  const status = results.reduce(
-    (curr, r) => (r.status >= 500 ? r.status : curr), 207,
-  );
+  const failure = results.find((r) => r.status >= 500);
+  if (failure) {
+    const e = new Error(failure.message);
+    e.status = failure.status;
+    e.path = failure.path;
+    throw e;
+  }
```

Comment:
If there was a recoverable failure, throw that instead of returning it as a `Response`